From 4aa658477841b6ac2edae83da9d9c8cc7788fa3a Mon Sep 17 00:00:00 2001
From: Ke Qinghua <qinghua.ke@freescale.com>
Date: Wed, 19 Mar 2014 11:09:53 +0800
Subject: [PATCH 300/304] ENGR00303453-1:[Android_ MX6DL_ARD]Boot: Android
 cannot boot up from NAND.  100%

Enable UBIFS selinux setting and label NFS, add untrusted_app access rule for cts test.

Signed-off-by: Ke Qinghua <qinghua.ke@freescale.com>
---
 evk_6sl/BoardConfig.mk                 | 5 ++++-
 evk_6sl/sepolicy/app.te                | 1 +
 evk_6sl/sepolicy/file_contexts         | 4 ++++
 evk_6sl/sepolicy/fs_use                | 4 ++++
 evk_6sl/sepolicy/genfs_contexts        | 1 +
 evk_6sl/sepolicy/untrusted_app.te      | 9 +++++++++
 sabreauto_6q/BoardConfig.mk            | 5 ++++-
 sabreauto_6q/sepolicy/app.te           | 1 +
 sabreauto_6q/sepolicy/file_contexts    | 4 ++++
 sabreauto_6q/sepolicy/fs_use           | 4 ++++
 sabreauto_6q/sepolicy/genfs_contexts   | 1 +
 sabreauto_6q/sepolicy/untrusted_app.te | 9 +++++++++
 sabresd_6dq/BoardConfig.mk             | 5 ++++-
 sabresd_6dq/sepolicy/app.te            | 1 +
 sabresd_6dq/sepolicy/file_contexts     | 4 ++++
 sabresd_6dq/sepolicy/fs_use            | 4 ++++
 sabresd_6dq/sepolicy/genfs_contexts    | 1 +
 sabresd_6dq/sepolicy/untrusted_app.te  | 9 +++++++++
 18 files changed, 69 insertions(+), 3 deletions(-)
 create mode 100644 evk_6sl/sepolicy/fs_use
 create mode 100644 evk_6sl/sepolicy/genfs_contexts
 create mode 100644 evk_6sl/sepolicy/untrusted_app.te
 create mode 100644 sabreauto_6q/sepolicy/fs_use
 create mode 100644 sabreauto_6q/sepolicy/genfs_contexts
 create mode 100644 sabreauto_6q/sepolicy/untrusted_app.te
 create mode 100644 sabresd_6dq/sepolicy/fs_use
 create mode 100644 sabresd_6dq/sepolicy/genfs_contexts
 create mode 100644 sabresd_6dq/sepolicy/untrusted_app.te

diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index 5849f83..dc5b70c 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -83,4 +83,7 @@ BOARD_SEPOLICY_DIRS := \
        device/fsl/evk_6sl/sepolicy
 BOARD_SEPOLICY_UNION := \
        app.te \
-       file_contexts
+       file_contexts \
+       fs_use \
+       untrusted_app.te \
+       genfs_contexts
diff --git a/evk_6sl/sepolicy/app.te b/evk_6sl/sepolicy/app.te
index 92d8fa6..158c7e3 100644
--- a/evk_6sl/sepolicy/app.te
+++ b/evk_6sl/sepolicy/app.te
@@ -2,3 +2,4 @@
 # They need that to render the standard UI.
 allow domain    graphics_device:chr_file rw_file_perms;
 allow domain    graphics_device:dir { write create add_name setattr mounton search };
+allow system    system_data_file:dir *;
diff --git a/evk_6sl/sepolicy/file_contexts b/evk_6sl/sepolicy/file_contexts
index ddb468a..55d7e38 100644
--- a/evk_6sl/sepolicy/file_contexts
+++ b/evk_6sl/sepolicy/file_contexts
@@ -1,2 +1,6 @@
 /system/bin/rtl_wpa_supplicant  u:object_r:wpa_exec:s0
 /system/bin/rtl_hostapd     u:object_r:hostapd_exec:s0
+/dev/ubi0_0                   u:object_r:block_device:s0
+/dev/ubi0_1                   u:object_r:block_device:s0
+/dev/ubi0_2                   u:object_r:block_device:s0
+/dev/ubi0_3                   u:object_r:block_device:s0
\ No newline at end of file
diff --git a/evk_6sl/sepolicy/fs_use b/evk_6sl/sepolicy/fs_use
new file mode 100644
index 0000000..848581b
--- /dev/null
+++ b/evk_6sl/sepolicy/fs_use
@@ -0,0 +1,4 @@
+# Label inodes via getxattr.
+fs_use_xattr ubifs u:object_r:labeledfs:s0;
+
+
diff --git a/evk_6sl/sepolicy/genfs_contexts b/evk_6sl/sepolicy/genfs_contexts
new file mode 100644
index 0000000..b27f218
--- /dev/null
+++ b/evk_6sl/sepolicy/genfs_contexts
@@ -0,0 +1 @@
+genfscon nfs / u:object_r:rootfs:s0
diff --git a/evk_6sl/sepolicy/untrusted_app.te b/evk_6sl/sepolicy/untrusted_app.te
new file mode 100644
index 0000000..d03f9c6
--- /dev/null
+++ b/evk_6sl/sepolicy/untrusted_app.te
@@ -0,0 +1,9 @@
+allow untrusted_app domain:dir r_dir_perms;
+allow untrusted_app domain:file rw_file_perms;
+allow untrusted_app domain:unix_stream_socket {read write};
+allow untrusted_app appdomain:dir r_dir_perms;
+allow untrusted_app appdomain:file rw_file_perms;
+allow untrusted_app fs_type:dir r_dir_perms;
+allow untrusted_app fs_type:file rw_file_perms;
+allow untrusted_app untrusted_app:process *;
+allow untrusted_app init_tmpfs:file rw_file_perms;
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index 5d379e8..f5cd4c0 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -108,4 +108,7 @@ BOARD_SEPOLICY_DIRS := \
 
 BOARD_SEPOLICY_UNION := \
        app.te \
-       file_contexts
+       file_contexts \
+       fs_use \
+       untrusted_app.te \
+       genfs_contexts
diff --git a/sabreauto_6q/sepolicy/app.te b/sabreauto_6q/sepolicy/app.te
index 92d8fa6..9418dbe 100755
--- a/sabreauto_6q/sepolicy/app.te
+++ b/sabreauto_6q/sepolicy/app.te
@@ -2,3 +2,4 @@
 # They need that to render the standard UI.
 allow domain    graphics_device:chr_file rw_file_perms;
 allow domain    graphics_device:dir { write create add_name setattr mounton search };
+allow system    system_data_file:dir *;
\ No newline at end of file
diff --git a/sabreauto_6q/sepolicy/file_contexts b/sabreauto_6q/sepolicy/file_contexts
index ddb468a..55d7e38 100644
--- a/sabreauto_6q/sepolicy/file_contexts
+++ b/sabreauto_6q/sepolicy/file_contexts
@@ -1,2 +1,6 @@
 /system/bin/rtl_wpa_supplicant  u:object_r:wpa_exec:s0
 /system/bin/rtl_hostapd     u:object_r:hostapd_exec:s0
+/dev/ubi0_0                   u:object_r:block_device:s0
+/dev/ubi0_1                   u:object_r:block_device:s0
+/dev/ubi0_2                   u:object_r:block_device:s0
+/dev/ubi0_3                   u:object_r:block_device:s0
\ No newline at end of file
diff --git a/sabreauto_6q/sepolicy/fs_use b/sabreauto_6q/sepolicy/fs_use
new file mode 100644
index 0000000..848581b
--- /dev/null
+++ b/sabreauto_6q/sepolicy/fs_use
@@ -0,0 +1,4 @@
+# Label inodes via getxattr.
+fs_use_xattr ubifs u:object_r:labeledfs:s0;
+
+
diff --git a/sabreauto_6q/sepolicy/genfs_contexts b/sabreauto_6q/sepolicy/genfs_contexts
new file mode 100644
index 0000000..b27f218
--- /dev/null
+++ b/sabreauto_6q/sepolicy/genfs_contexts
@@ -0,0 +1 @@
+genfscon nfs / u:object_r:rootfs:s0
diff --git a/sabreauto_6q/sepolicy/untrusted_app.te b/sabreauto_6q/sepolicy/untrusted_app.te
new file mode 100644
index 0000000..d03f9c6
--- /dev/null
+++ b/sabreauto_6q/sepolicy/untrusted_app.te
@@ -0,0 +1,9 @@
+allow untrusted_app domain:dir r_dir_perms;
+allow untrusted_app domain:file rw_file_perms;
+allow untrusted_app domain:unix_stream_socket {read write};
+allow untrusted_app appdomain:dir r_dir_perms;
+allow untrusted_app appdomain:file rw_file_perms;
+allow untrusted_app fs_type:dir r_dir_perms;
+allow untrusted_app fs_type:file rw_file_perms;
+allow untrusted_app untrusted_app:process *;
+allow untrusted_app init_tmpfs:file rw_file_perms;
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 43d1c0f..3115f26 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -114,4 +114,7 @@ BOARD_SEPOLICY_DIRS := \
 
 BOARD_SEPOLICY_UNION := \
        app.te \
-       file_contexts
+       file_contexts \
+       fs_use \
+       untrusted_app.te \
+       genfs_contexts
diff --git a/sabresd_6dq/sepolicy/app.te b/sabresd_6dq/sepolicy/app.te
index 92d8fa6..9418dbe 100644
--- a/sabresd_6dq/sepolicy/app.te
+++ b/sabresd_6dq/sepolicy/app.te
@@ -2,3 +2,4 @@
 # They need that to render the standard UI.
 allow domain    graphics_device:chr_file rw_file_perms;
 allow domain    graphics_device:dir { write create add_name setattr mounton search };
+allow system    system_data_file:dir *;
\ No newline at end of file
diff --git a/sabresd_6dq/sepolicy/file_contexts b/sabresd_6dq/sepolicy/file_contexts
index ddb468a..55d7e38 100644
--- a/sabresd_6dq/sepolicy/file_contexts
+++ b/sabresd_6dq/sepolicy/file_contexts
@@ -1,2 +1,6 @@
 /system/bin/rtl_wpa_supplicant  u:object_r:wpa_exec:s0
 /system/bin/rtl_hostapd     u:object_r:hostapd_exec:s0
+/dev/ubi0_0                   u:object_r:block_device:s0
+/dev/ubi0_1                   u:object_r:block_device:s0
+/dev/ubi0_2                   u:object_r:block_device:s0
+/dev/ubi0_3                   u:object_r:block_device:s0
\ No newline at end of file
diff --git a/sabresd_6dq/sepolicy/fs_use b/sabresd_6dq/sepolicy/fs_use
new file mode 100644
index 0000000..848581b
--- /dev/null
+++ b/sabresd_6dq/sepolicy/fs_use
@@ -0,0 +1,4 @@
+# Label inodes via getxattr.
+fs_use_xattr ubifs u:object_r:labeledfs:s0;
+
+
diff --git a/sabresd_6dq/sepolicy/genfs_contexts b/sabresd_6dq/sepolicy/genfs_contexts
new file mode 100644
index 0000000..b27f218
--- /dev/null
+++ b/sabresd_6dq/sepolicy/genfs_contexts
@@ -0,0 +1 @@
+genfscon nfs / u:object_r:rootfs:s0
diff --git a/sabresd_6dq/sepolicy/untrusted_app.te b/sabresd_6dq/sepolicy/untrusted_app.te
new file mode 100644
index 0000000..d03f9c6
--- /dev/null
+++ b/sabresd_6dq/sepolicy/untrusted_app.te
@@ -0,0 +1,9 @@
+allow untrusted_app domain:dir r_dir_perms;
+allow untrusted_app domain:file rw_file_perms;
+allow untrusted_app domain:unix_stream_socket {read write};
+allow untrusted_app appdomain:dir r_dir_perms;
+allow untrusted_app appdomain:file rw_file_perms;
+allow untrusted_app fs_type:dir r_dir_perms;
+allow untrusted_app fs_type:file rw_file_perms;
+allow untrusted_app untrusted_app:process *;
+allow untrusted_app init_tmpfs:file rw_file_perms;
-- 
1.8.0

